--- origsrc/rep-gtk-0.18.4/Makefile.in	2009-03-05 13:59:04.454456000 -0600
+++ src/rep-gtk-0.18.4/Makefile.in	2009-03-05 13:59:26.110567400 -0600
@@ -22,10 +22,10 @@ CC=@CC@
 CPPFLAGS=@CPPFLAGS@
 CFLAGS=@CFLAGS@ @DEVELOPMENT_CFLAGS@
 LDFLAGS=@LDFLAGS@
-override CPPFLAGS += -DHAVE_CONFIG_H -I.
+override CPPFLAGS += -DHAVE_CONFIG_H -I. -I$(top_srcdir)
 override CFLAGS += @GLIB_CFLAGS@ @GTK_CFLAGS@ @REP_CFLAGS@ @GDK_PIXBUF_CFLAGS@
-REP_SGTK_LIBS=@GLIB_LIBS@ @LIBS@
-REP_GTK_LIBS=@GTK_LIBS@ @LIBS@
+REP_SGTK_LIBS=@GLIB_LIBS@ @REP_LIBS@ @LIBS@
+REP_GTK_LIBS=@GTK_LIBS@ @REP_LIBS@ @LIBS@
 
 INSTALL=@INSTALL@
 INSTALL_DATA=@INSTALL_DATA@
@@ -47,11 +47,11 @@ all : build_dirs $(OUT)
 gtk-2/types.la : $(SGTK_OBJS)
 	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_SGTK_LIBS)
 
-gtk-2/gtk.la : $(OBJS)
+gtk-2/gtk.la : $(OBJS) gtk-2/types.la
 	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GTK_LIBS)
 
 %-glue.c : %.defs build-gtk.jlc
-	rep build-gtk.jlc -f build-gtk-batch $< $@
+	rep $(top_srcdir)/build-gtk.jlc -f build-gtk-batch $< $@
 
 build-gtk.jlc : build-gtk.jl
 	rep compiler -f compile-batch $<
--- origsrc/rep-gtk-0.18.4/configure.in	2009-02-02 03:15:02.000000000 -0600
+++ src/rep-gtk-0.18.4/configure.in	2009-03-05 13:56:41.252304700 -0600
@@ -49,6 +49,7 @@ REP_MIN_VER="0.17.3"
 PKG_CHECK_MODULES(LIBREP, librep >= $REP_MIN_VER
 		  ,REP_EXECDIR="`pkg-config --variable=repcommonexecdir librep`"
 		  REP_CFLAGS="`pkg-config --cflags librep`"
+		  REP_LIBS="`pkg-config --libs librep`"
 		  ,AC_MSG_ERROR([can't locate librep]))
 
 dnl Check for glib
@@ -86,6 +87,7 @@ AC_SUBST(DEVELOPMENT_CFLAGS)
 AC_SUBST(LIBS)
 AC_SUBST(REP_EXECDIR)
 AC_SUBST(REP_CFLAGS)
+AC_SUBST(REP_LIBS)
 AC_SUBST(REP_MIN_VER)
 AC_SUBST(GTK_MIN_VER)
 AC_SUBST(GLIB_MIN_VER)
